packages/@ember/-internals/runtime/lib/mixins/array.js

```diff
@@ -564,8 +565,10 @@ const ArrayMixin = Mixin.create(Enumerable, {
     @property {Boolean} hasArrayObservers
     @public
   */
-  hasArrayObservers: nonEnumerableComputed(function() {
-    return hasListeners(this, '@array:change') || hasListeners(this, '@array:before');
+  hasArrayObservers: descriptor({
```

Comment:
Don’t we also need to add enumerable: false? Also, is caching important here?

Reply:
Ah, yes, I meant to do that! 

For caching, I think it depends on how many times we expect folks to run `addArrayObserver` and `removeArrayObserver`, since nothing else really [uses it in the ecosystem](https://www.emberobserver.com/code-search?codeQuery=hasArrayObserver) or the codebase. It seems like that's [also a relatively uncommon thing](https://www.emberobserver.com/code-search?codeQuery=addArrayObserver), but we could also keep it as a CP if we're worried and use `untrack` to access it and prevent the issue.